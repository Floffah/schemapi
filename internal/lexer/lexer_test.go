package lexer

import (
	"os"
	"strings"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestLexerSnapshot(t *testing.T) {
	files, err := os.ReadDir("../../tests")

	if err != nil {
		t.Fatalf("failed to read examples directory: %v", err)
	}

	for _, file := range files {
		if !strings.HasSuffix(file.Name(), ".sapi") {
			continue
		}

		t.Run(file.Name(), func(t *testing.T) {
			data, err := os.ReadFile("../../tests/" + file.Name())

			if err != nil {
				t.Fatalf("failed to read %s: %v", file.Name(), err)
			}

			lexer := NewLexer(string(data))
			tokens := lexer.Lex()

			snaps.MatchSnapshot(t, tokens)
		})
	}
}
