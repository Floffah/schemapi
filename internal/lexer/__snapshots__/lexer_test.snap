[TestLexerSnapshot/callable.sapi - 1]
[]lexer.Token{
    {Type:"Keyword", Value:"callable", Line:1, Col:9},
    {Type:"Ident", Value:"http.get", Line:1, Col:18},
    {Type:"Path", Value:"/healthz", Line:1, Col:27},
    {Type:"LBrace", Value:"{", Line:2, Col:0},
    {Type:"RBrace", Value:"}", Line:2, Col:2},
}
---

[TestLexerSnapshot/full.sapi - 1]
[]lexer.Token{
    {Type:"Keyword", Value:"callable", Line:1, Col:9},
    {Type:"Ident", Value:"http.get", Line:1, Col:18},
    {Type:"Path", Value:"/healthz", Line:1, Col:27},
    {Type:"LBrace", Value:"{", Line:2, Col:0},
    {Type:"Ident", Value:"query", Line:2, Col:8},
    {Type:"Ident", Value:"string", Line:2, Col:15},
    {Type:"String", Value:"name", Line:2, Col:22},
    {Type:"LBrace", Value:"{", Line:3, Col:0},
    {Type:"Ident", Value:"description", Line:3, Col:16},
    {Type:"Equal", Value:"=", Line:3, Col:18},
    {Type:"String", Value:"Name of the service to check health for", Line:4, Col:0},
    {Type:"Ident", Value:"default", Line:4, Col:12},
    {Type:"Equal", Value:"=", Line:4, Col:14},
    {Type:"String", Value:"default-service", Line:5, Col:0},
    {Type:"RBrace", Value:"}", Line:6, Col:0},
    {Type:"Ident", Value:"response", Line:7, Col:11},
    {Type:"String", Value:"OK", Line:7, Col:16},
    {Type:"LBrace", Value:"{", Line:8, Col:0},
    {Type:"Ident", Value:"status", Line:8, Col:8},
    {Type:"Equal", Value:"=", Line:8, Col:10},
    {Type:"Number", Value:"200", Line:9, Col:0},
    {Type:"Ident", Value:"body", Line:9, Col:6},
    {Type:"Equal", Value:"=", Line:9, Col:8},
    {Type:"LBrace", Value:"{", Line:10, Col:0},
    {Type:"Ident", Value:"message", Line:10, Col:11},
    {Type:"Equal", Value:"=", Line:10, Col:13},
    {Type:"String", Value:"Service {name} is healthy", Line:11, Col:0},
    {Type:"RBrace", Value:"}", Line:12, Col:0},
    {Type:"RBrace", Value:"}", Line:13, Col:0},
    {Type:"Ident", Value:"response", Line:14, Col:11},
    {Type:"String", Value:"Service Not Found", Line:14, Col:31},
    {Type:"LBrace", Value:"{", Line:15, Col:0},
    {Type:"Ident", Value:"status", Line:15, Col:10},
    {Type:"Equal", Value:"=", Line:15, Col:12},
    {Type:"Number", Value:"404", Line:16, Col:0},
    {Type:"Ident", Value:"body", Line:16, Col:8},
    {Type:"Equal", Value:"=", Line:16, Col:10},
    {Type:"LBrace", Value:"{", Line:17, Col:0},
    {Type:"Ident", Value:"message", Line:17, Col:13},
    {Type:"Equal", Value:"=", Line:17, Col:15},
    {Type:"String", Value:"Service {name} not found", Line:18, Col:0},
    {Type:"RBrace", Value:"}", Line:19, Col:0},
    {Type:"RBrace", Value:"}", Line:20, Col:0},
    {Type:"Ident", Value:"response", Line:21, Col:11},
    {Type:"String", Value:"Internal Server Error", Line:21, Col:35},
    {Type:"LBrace", Value:"{", Line:22, Col:0},
    {Type:"Ident", Value:"status", Line:22, Col:8},
    {Type:"Equal", Value:"=", Line:22, Col:10},
    {Type:"Number", Value:"500", Line:23, Col:0},
    {Type:"Ident", Value:"body", Line:23, Col:6},
    {Type:"Equal", Value:"=", Line:23, Col:8},
    {Type:"LBrace", Value:"{", Line:24, Col:0},
    {Type:"Ident", Value:"message", Line:24, Col:11},
    {Type:"Equal", Value:"=", Line:24, Col:13},
    {Type:"String", Value:"An unexpected error occurred", Line:25, Col:0},
    {Type:"RBrace", Value:"}", Line:26, Col:0},
    {Type:"RBrace", Value:"}", Line:27, Col:0},
    {Type:"RBrace", Value:"}", Line:27, Col:2},
}
---
